listenbrainz/webserver/static/js/jsx/api-service.js

```diff
@@ -0,0 +1,89 @@
+import {isFinite, isNil, isString} from 'lodash';
+
+export default class APIService {
+  
+  APIBaseURI;
+
+  constructor(APIBaseURI){
+    if(isNil(APIBaseURI) || !isString(APIBaseURI)){
+      throw new SyntaxError(`Expected API base URI string, got ${typeof APIBaseURI} instead`)
+    }
+    if(APIBaseURI.endsWith('/')){
+      APIBaseURI = APIBaseURI.substring(0, APIBaseURI.length-1);
+    }
+    if(!APIBaseURI.endsWith('/1')){
+      APIBaseURI += '/1';
+    }
+    this.APIBaseURI = APIBaseURI;
+  }
+
+  async getRecentListensForUsers(userNames, limit) {
+    let userNamesForQuery = userNames;
```

Comment:
I've gotten used to follow some eslint defaults, including one that considers it bad practice to reassign function arguments (`userNames = userNames.join…`).
It's for clarity of code rather than anything else.

Reply:
You'll like the name: https://eslint.org/docs/rules/no-param-reassign

Reply:
oooh, makes sense. the initial assignment to `userNames` threw me off a bit.